src/typeahead/typeahead.ts

```diff
@@ -357,9 +358,15 @@ export class NgbTypeahead implements ControlValueAccessor,
         // the change detection turn wouldn't be invoked automatically.
         this._windowRef.changeDetectorRef.detectChanges();
       }
+
+      this._live.say(this._srOptionsSummaryTemplate({count: results.length}));
```

Comment:
Why do you pass an object here instead of just a number? It results in the unnecessary object creation and you read only `count` from it anyway.

Reply:
or no need in method at all

Reply:
_I don't know why this appears as an outdated comment, but I'll reply anyways_

No need for that indeed. I started with a public API, so I made it generic:

- users would give a callback, which is now the private method we can see
- I would give them an object since this is more future-proof in case we want to give more information

I kept this design while making things private.

However I would not be concerned __AT ALL__ with performances. These are micro-optimizations, and I think you are well placed to know that actual performance is more complex than that. In this context, the feature call rate is __way too low__ to worry about extra nanoseconds spent creating an object, calling a method and reading from this object. My concerns would be more about maintainability. But here I can remove both the method and the object as long as it remains private, it's a matter of one line of code so it's perfectly fine.